python/ray/_private/accelerators/__init__.py

```diff
@@ -29,25 +29,34 @@ def get_all_accelerator_resource_names() -> Set[str]:
     }
 
 
-_resource_name_to_accelerator_manager = {
-    accelerator_manager.get_resource_name(): accelerator_manager
-    for accelerator_manager in get_all_accelerator_managers()
-}
-
-
-def get_accelerator_manager_for_resource(resource_name: str) -> AcceleratorManager:
+def get_accelerator_manager_for_resource(
+    resource_name: str,
+) -> Optional[AcceleratorManager]:
     """Get the corresponding accelerator manager for the given
     accelerator resource name
 
     E.g., TPUAcceleratorManager is returned if resource name is "TPU"
     """
-    if resource_name == "GPU":
+    try:
+        return get_accelerator_manager_for_resource._resource_name_to_accelerator_manager.get(  # noqa: E501
+            resource_name, None
+        )
+    except AttributeError:
+        # Lazy initialization.
+        resource_name_to_accelerator_manager = {
```

Comment:
oh this gets cache across invocation? i thought function doesn't have attribute cached like this.

Reply:
Yes, this basically implements function static variable in C++.